fs/nova/balloc.c

```diff
@@ -627,6 +627,11 @@ bool nova_alloc_superpage(struct super_block *sb,
 		 *    requested blocks.
 		 * Otherwise, we are free to go.
 		 */
+		if (num_blocks > (curr_blocks - left_margin)) {
```

Comment:
Add a comment that we are doing best effort here to allocate as many 2MB blocks as possible.